internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -112,10 +116,32 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 	// NOTE: Once we reconcile node labels for the first time, the NodeUninitializedTaint is removed from the node.
 	nodeLabels := getManagedLabels(machine.Labels)
 
+	// Get interruptible instance status from the infrastructure provider and set the interruptible label on the node.
+	interruptible := false
+	found := false
+	if infraMachine != nil {
+		interruptible, found, err = unstructured.NestedBool(infraMachine.Object, "status", "interruptible")
+		if err != nil {
+			return ctrl.Result{}, errors.Wrapf(err, "failed to get status interruptible from infra machine %s", klog.KObj(infraMachine))
+		}
+		// If interruptible is set and is true add the interruptible label to the node labels.
+		if found && interruptible {
+			nodeLabels[clusterv1.InterruptibleLabel] = ""
+		}
+	}
+
+	_, nodeHadInterruptibleLabel := node.Labels[clusterv1.InterruptibleLabel]
```

Comment:
q: should this be computed before setting the interruptible label (before L122)?

Reply:
The label is set on nodeLabels not node.Labels

Reply:
`node.Labels` as not yet updated here. `node.Labels` is updated after `patchNode` is called in l.136.